File: GARI_Umbria_2026-08-03/GiasNG/src/app/Model/anagrafiche/CostoUnitario.ts
import { UnitaDiMisura } from '../metaschema/UnitaDiMisura';
import { IntervalloTemporale } from './IntervalloTemporale';

export class CostoUnitario {
    codice: number;
    unitaDiMisura: UnitaDiMisura;
    prezzo: number;
    validita: IntervalloTemporale;
    flag_cancellazione: boolean;

    constructor() {
        this.flag_cancellazione = false;
    }
}

export class CostoUnitarioChiave extends CostoUnitario {
    chiave = 0;
}

File: GARI_Umbria_2026-08-03/DotNet/AgronicaCoreStd/AgronicaCoreModelsSTD/anagrafiche/CostoUnitario.cs
using AgronicaCoreModelsSTD.metaschema;
using System;
using System.Collections.Generic;
using System.Text;

namespace AgronicaCoreModelsSTD.anagrafiche
{
    public class CostoUnitario
    {

        public int codice { get; set; }

        public UnitaDiMisura unitaDiMisura { get; set; }

        public double prezzo { get; set; }

        public IntervalloTemporale validita { get; set; }

        public bool flag_cancellazione { get; set; }

        public CostoUnitario(int codice)
        {
            this.codice = codice;
            this.flag_cancellazione = false;
        }

        public CostoUnitario() { }

    }
}
